### PR TITLE
Added Insertion Sort in C#

### LINKED ISSUE #633: Insertion Sort in C#
This is an open issue for the Insertion Sort algorithm implementation in C#. If you wish to contribute, just make some noise and we will assign you.

### PR DESCRIPTION
**Pull request by riturajjain2000**
**Fixes #633**

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [x] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [x] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [x] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [x] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [x] This pull request shall only be merged once the build passes.

-   [x] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -146,6 +146,10 @@ The community maintained a list of **Algorithms and Data Structures** implementa
       <td><code>gr-dij</code></td>
       <td class="text-center">&#10004;</td>
       <td class="text-center">&#10004;</td>
+<<<<<<< HEAD
```
Comment (RavinduSachintha) [inline README.md:149]: please check the changes again. these parts are some remaining texts from conflicts. you have to remove them.

**Review by RavinduSachintha — CHANGES_REQUESTED** [review]

**Review by RavinduSachintha — APPROVED** [review]

**Comment by RavinduSachintha** [comment]
@riturajjain2000 would you be able to add this contribution to the algorithm chart in the main readme file? don't change anything else. :+1:

**Comment by riturajjain2000** [comment]
> @riturajjain2000 would you be able to add this contribution to the algorithm chart in the main readme file? don't change anything else. 👍

Done

**Review by RavinduSachintha — APPROVED** [review]

**Comment by RavinduSachintha** [comment]
@riturajjain2000 thank you for your contribution. Keep it up :+1:

**Comment by riturajjain2000** [comment]
Happy to help 😄

**Pull request merged**